### PR TITLE
Fix Gavin app (#5124)/VCF import/annotator in compound(#5120)

### PR DESCRIPTION
**Pull request by bartcharbon**
#### Checklist
- [ ] Unit tests
- [ ] Updated testplan

### CODE REVIEW

File: molgenis-gavin/src/test/java/org/molgenis/gavin/job/FailingAnnotator.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+	@Override
+	public CmdLineAnnotatorSettingsConfigurer getCmdLineAnnotatorSettingsConfigurer()
+	{
+		return new CmdLineAnnotatorSettingsConfigurer()
+		{
+			@Override
+			public void addSettings(String annotationSourceFileName)
+			{
+
```
Comment (jjettenn) [inline molgenis-gavin/src/test/java/org/molgenis/gavin/job/FailingAnnotator.java]: No setting is added. As intended?

File: molgenis-gavin/src/test/java/org/molgenis/gavin/job/FailingAnnotator.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	@Override
+	public String getSimpleName()
+	{
+		return "FAIL";
+	}
+
+	@Override
+	public String getFullName()
+	{
+		return "FAIL";
```
Comment (jjettenn) [inline molgenis-gavin/src/test/java/org/molgenis/gavin/job/FailingAnnotator.java]: no package?

File: molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/core/utils/AnnotatorUtils.java [hunk 3]
```diff
@@ -63,14 +80,138 @@ else if (StringUtils.isEmpty(annotations))
...
+					.setDataType(MolgenisFieldTypes.AttributeType.COMPOUND).setLabel(annotator.getSimpleName());
+			AttributeMetaData finalCompound = compound;
+			attributeMetaDatas.stream().filter(part -> entityMetaData.getAttribute(part.getName()) == null)
+					.forEachOrdered(part -> finalCompound.addAttributePart(part));
+			entityMetaData.addAttribute(compound);
+		}
 		return entityMetaData;
 	}
+
+	public static String annotate(RepositoryAnnotator annotator, VcfAttributes vcfAttributes,
```
Comment (jjettenn) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/core/utils/AnnotatorUtils.java]: This is a very long method can you please cut it into smaller functions and add comments?

File: molgenis-annotators-cmd/src/main/java/org/molgenis/annotation/cmd/CmdLineAnnotator.java [hunk 4]
```diff
@@ -171,7 +156,33 @@ else if (inputVcfFile.isDirectory())
...
+	 */
+	private void annotate(RepositoryAnnotator annotator, VcfAttributes vcfAttributes,
+			EntityMetaDataFactory entityMetaDataFactory, AttributeMetaDataFactory attributeMetaDataFactory,
+			VcfUtils vcfUtils, File inputVcfFile, File outputVCFFile, OptionSet options) throws Exception
+	{
+		List<String> attributesToInclude = options.nonOptionArguments().stream().map(Object::toString)
+				.collect(Collectors.toList());
+		AnnotatorUtils.annotate(annotator, vcfAttributes, entityMetaDataFactory, attributeMetaDataFactory, vcfUtils,
+				inputVcfFile, outputVCFFile, attributesToInclude, options.has("u"));
+		if (options.has("validate"))
```
Comment (jjettenn) [inline molgenis-annotators-cmd/src/main/java/org/molgenis/annotation/cmd/CmdLineAnnotator.java]: Change validate to a constant or use an existing constant

File: molgenis-gavin/src/test/java/org/molgenis/gavin/job/FailingAnnotator.java (hunk 1 above)
Comment (bartcharbon) [inline molgenis-gavin/src/test/java/org/molgenis/gavin/job/FailingAnnotator.java]: yes, it is only used for the unit test

File: molgenis-gavin/src/test/java/org/molgenis/gavin/job/FailingAnnotator.java (hunk 2 above)
Comment (bartcharbon) [inline molgenis-gavin/src/test/java/org/molgenis/gavin/job/FailingAnnotator.java]: no, only used for unit test

File: molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/core/utils/AnnotatorUtils.java (hunk 3 above)
Comment (bartcharbon) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/core/utils/AnnotatorUtils.java]: this is a bugfix, I don't think refactoring existing code is in scope

File: molgenis-annotators-cmd/src/main/java/org/molgenis/annotation/cmd/CmdLineAnnotator.java (hunk 4 above)
Comment (bartcharbon) [inline molgenis-annotators-cmd/src/main/java/org/molgenis/annotation/cmd/CmdLineAnnotator.java]: this is a bugfix, I don't think refactoring exisiting code is in scope

**Pull request merged**